UnityAds/src/main/java/com/mopub/mobileads/UnityRewardedVideo.java

```diff
@@ -90,7 +92,25 @@ protected void load(@NonNull final Context context, @NonNull final AdData adData
 
         setAutomaticImpressionAndClickTracking(false);
 
-        UnityAds.load(mPlacementId);
+        UnityAds.load(mPlacementId, new IUnityAdsLoadListener() {
+            @Override
+            public void onUnityAdsAdLoaded(String placementId) {
+                MoPubLog.log(CUSTOM, ADAPTER_NAME, "Unity rewarded video cached for placement " + placementId + ".");
+                if (mLoadListener != null) {
+                    mLoadListener.onAdLoaded();
+                }
+                MoPubLog.log(LOAD_SUCCESS, ADAPTER_NAME);
+            }
+
+            @Override
+            public void onUnityAdsFailedToLoad(String placementId) {
+                if (mLoadListener != null) {
+                    mLoadListener.onAdLoadFailed(MoPubErrorCode.NO_FILL);
+                }
+                MoPubLog.log(LOAD_FAILED, ADAPTER_NAME, MoPubErrorCode.NETWORK_NO_FILL.getIntCode(), MoPubErrorCode.NETWORK_NO_FILL);
+                UnityRouter.getInterstitialRouter().removeListener(mPlacementId);
```

Comment:
where we added listener? do we still need to remove?

Reply:
I think so, since it is failed to load, we can remove this listener